Mi22019

spaske00 opened the pull request:
Poštovani,

Prilažem sadašnju verziju projekta iz Računarske grafike. Još uvek nisam siguran da li je finalna, s obzirom da mi je ostala neimplementirana lekcija iz grupe B, na šta ću se možda vratiti do kraja semestra.

Ova verzija projekta sadrži osnovne detalje implementacije, kao i temu iz grupe A (instanciranje).

Ovim putem bih želeo da proverim kvalitet projekta, kao i broj poena koje bih osvojio na sadašnjoj verziji projekta.

Srdačno,
David Ćuruvija 19/2022

davidcuruvija committed:
Created Empty App and app_setup
Created base for MainController

davidcuruvija committed:
Created Empty App and app_setup
Created base for MainController
Added floor

davidcuruvija committed:
Implemented camera movement and rotation
Added floor

davidcuruvija committed:
Project Topic Changed
Added multiple graves

davidcuruvija committed:
Project Topic Changed
Added multiple graves

davidcuruvija committed:
Changed floor to cobblestone - it is more realistic, the first one was just a test

davidcuruvija committed:
Added GUIController
Added lamp

davidcuruvija committed:
Added a dog and a lantern

davidcuruvija committed:
Added a car

davidcuruvija committed:
Added lighting for the lamp and the lantern

davidcuruvija committed:
Implemented sequence of actions:
Press key B -> wait 2 seconds -> lamp color changes to green -> wait 2 seconds -> lamp color changes to blue -> wait 3 seconds -> lamp color changes to normal (transparent)

davidcuruvija committed:
Implemented manually changing colors of lamp on key N

davidcuruvija committed:
Fixed lantern
Added new Gull Model
Implemented instancing for gulls and graves (TEMA IZ GRUPE A)

davidcuruvija committed:
Fixed lamp light

davidcuruvija committed:
Fixed lamp light

davidcuruvija committed:
Removed car

davidcuruvija committed:
Added useful comments for some actions
The project has been finalized and is available for review

davidcuruvija committed:
PROJECT-DESCRIPTION-TEMPLATE.md popunjen

spaske00 commented:
Projekat se ne bilduje. Molim Vas, popravite grešku i javite mi da pogledam ponovo. 
Izlaz na terminalu:

```
====================[ Build | APP | Debug ]=====================================
/home/ms/Programs/clion-2024.3.1/bin/cmake/linux/x64/bin/cmake --build /home/ms/Desktop/RGProjects/davidcuruvija/cmake-build-debug --target APP -j 14
[277/313] Building C object engine/libs/assimp/code/CMakeFiles/assimp.dir/__/contrib/zip/src/zip.c.o
In file included from /home/ms/Desktop/RGProjects/davidcuruvija/engine/libs/assimp/contrib/zip/src/zip.c:44:
/home/ms/Desktop/RGProjects/davidcuruvija/engine/libs/assimp/contrib/zip/src/miniz.h:5108:9: note: ‘#pragma message: Using fopen, ftello, fseeko, stat() etc. path for file I/O - this path may not support large files.’
 5108 | #pragma message(                                                               \
      |         ^~~~~~~
[286/313] Building CXX object engine/libs/imgui/CMakeFiles/imgui.dir/src/imgui_widgets.cpp.o
/home/ms/Desktop/RGProjects/davidcuruvija/engine/libs/imgui/src/imgui_widgets.cpp: In function ‘bool ImGui::CollapsingHeader(const char*, bool*, ImGuiTreeNodeFlags)’:
/home/ms/Desktop/RGProjects/davidcuruvija/engine/libs/imgui/src/imgui_widgets.cpp:5897:54: warning: bitwise operation between different enumeration types ‘ImGuiTreeNodeFlags_’ and ‘ImGuiTreeNodeFlagsPrivate_’ is deprecated [-Wdeprecated-enum-enum-conversion]
 5897 |         flags |= ImGuiTreeNodeFlags_AllowItemOverlap | ImGuiTreeNodeFlags_ClipLabelForTrailingButton;
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[295/313] Building CXX object app/CMakeFiles/APP.dir/src/GraveyardSimulation.cpp.o
FAILED: app/CMakeFiles/APP.dir/src/GraveyardSimulation.cpp.o 
/usr/bin/g++-13 -DIMGUI_IMPL_OPENGL_LOADER_GLAD -DSPDLOG_COMPILED_LIB -I/home/ms/Desktop/RGProjects/davidcuruvija/app/include -I/home/ms/Desktop/RGProjects/davidcuruvija/engine/include -I/home/ms/Desktop/RGProjects/davidcuruvija/engine/libs/glm -I/home/ms/Desktop/RGProjects/davidcuruvija/engine/libs/spdlog/include -I/home/ms/Desktop/RGProjects/davidcuruvija/engine/libs/imgui/include -I/home/ms/Desktop/RGProjects/davidcuruvija/engine/libs/json/include -g -std=c++23 -fdiagnostics-color=always -MD -MT app/CMakeFiles/APP.dir/src/GraveyardSimulation.cpp.o -MF app/CMakeFiles/APP.dir/src/GraveyardSimulation.cpp.o.d -o app/CMakeFiles/APP.dir/src/GraveyardSimulation.cpp.o -c /home/ms/Desktop/RGProjects/davidcuruvija/app/src/GraveyardSimulation.cpp
In file included from /home/ms/Desktop/RGProjects/davidcuruvija/app/include/../../engine/libs/assimp/code/AssetLib/Blender/BlenderScene.h:49,
                 from /home/ms/Desktop/RGProjects/davidcuruvija/app/include/LampEvent.hpp:7,
                 from /home/ms/Desktop/RGProjects/davidcuruvija/app/include/../include/MainController.hpp:8,
                 from /home/ms/Desktop/RGProjects/davidcuruvija/app/src/GraveyardSimulation.cpp:8:
/home/ms/Desktop/RGProjects/davidcuruvija/app/include/../../engine/libs/assimp/code/AssetLib/Blender/BlenderDNA.h:50:10: fatal error: assimp/BaseImporter.h: No such file or directory
   50 | #include <assimp/BaseImporter.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
[296/313] Building CXX object app/CMakeFiles/APP.dir/include/MainController.cpp.o
FAILED: app/CMakeFiles/APP.dir/include/MainController.cpp.o 
/usr/bin/g++-13 -DIMGUI_IMPL_OPENGL_LOADER_GLAD -DSPDLOG_COMPILED_LIB -I/home/ms/Desktop/RGProjects/davidcuruvija/app/include -I/home/ms/Desktop/RGProjects/davidcuruvija/engine/include -I/home/ms/Desktop/RGProjects/davidcuruvija/engine/libs/glm -I/home/ms/Desktop/RGProjects/davidcuruvija/engine/libs/spdlog/include -I/home/ms/Desktop/RGProjects/davidcuruvija/engine/libs/imgui/include -I/home/ms/Desktop/RGProjects/davidcuruvija/engine/libs/json/include -g -std=c++23 -fdiagnostics-color=always -MD -MT app/CMakeFiles/APP.dir/include/MainController.cpp.o -MF app/CMakeFiles/APP.dir/include/MainController.cpp.o.d -o app/CMakeFiles/APP.dir/include/MainController.cpp.o -c /home/ms/Desktop/RGProjects/davidcuruvija/app/include/MainController.cpp
In file included from /home/ms/Desktop/RGProjects/davidcuruvija/app/include/../../engine/libs/assimp/code/AssetLib/Blender/BlenderScene.h:49,
                 from /home/ms/Desktop/RGProjects/davidcuruvija/app/include/LampEvent.hpp:7,
                 from /home/ms/Desktop/RGProjects/davidcuruvija/app/include/MainController.hpp:8,
                 from /home/ms/Desktop/RGProjects/davidcuruvija/app/include/MainController.cpp:5:
/home/ms/Desktop/RGProjects/davidcuruvija/app/include/../../engine/libs/assimp/code/AssetLib/Blender/BlenderDNA.h:50:10: fatal error: assimp/BaseImporter.h: No such file or directory
   50 | #include <assimp/BaseImporter.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
[308/313] Building CXX object engine/libs/assimp/code/CMakeFiles/assimp.dir/AssetLib/IFC/IFCReaderGen1_2x3.cpp.o
ninja: build stopped: subcommand failed.
```

davidcuruvija committed:
Error fixed

davidcuruvija commented:
Poštovani,

Popravio sam greške, nadam se da je sad sve u redu.

spaske00 commented:
Projektu fali model `GULL.obj`. Ne vidim
Molim Vas, postavite i model na trenutnu granu. Ukoliko je model prevelik za git, možete ga okačiti na Vaš gugl drajv i ostaviti link u opisu projekta.
```
[2025-07-21 20:01:31.601] [info] load_model(name=gull, path=resources/models/gull/GULL.obj)
[2025-07-21 20:01:31.601] [error] AssetLoadingError: in /home/ms/Desktop/RGProjects/davidcuruvija/engine/src/ResourcesController.cpp:136.
 Message: Assimp error while reading model: resources/models/gull/GULL.obj from path gull.
```

davidcuruvija committed:
Added missing files

davidcuruvija committed:
Added missing files

davidcuruvija commented:
Dodao sam nedostajuće fajlove (drugi commit), iz nekog razloga se nisu ubacili prilikom prethodnih commit-ova.

davidcuruvija committed:
Added missing files

davidcuruvija commented:
Ipak su se tek sada dodali.

spaske00 commented on app/include/GuiController.cpp:
```diff
@@ -0,0 +1,28 @@
+#include <engine/graphics/GraphicsController.hpp>
+#include <engine/platform/PlatformController.hpp>
+
+namespace app {
```
[REFACTOR] Prebacite .cpp fajlove iz include/ u src/

spaske00 commented on app/include/MainController.cpp:
```diff
@@ -0,0 +1,278 @@
+
+    for (int i = 0; i < instanceCount; i++) {
+        float angle = (float(i) / instanceCount) * 2.0f * glm::pi<float>();
+        float x = centerX + radiusX * cos(angle) + (rand() % 1000 / 1000.0f - 0.5f) * 5.0f;
```
[REFACTOR] koristiti zaglavlje <random> umesto rand() za generisanje slučajnih vrednosti

spaske00 commented on PROJECT-DESCRIPTION-TEMPLATE.md:
```diff
@@ -1,31 +1,44 @@
-[ ] {ACTION_X} --- AFTER_M_SECONDS---Triggers---> {EVENT_A} ---> AFTER_N_SECONDS---Triggers---> {EVENT_B}
+[ + ] Model with lighting
+[ + ] Two types of lighting with customizable colors and movement through GUI or ACTIONS
+[ + ] {ACTION_X} --- AFTER_M_SECONDS---Triggers---> {EVENT_A} ---> AFTER_N_SECONDS---Triggers---> {EVENT_B}
```
Popuniti opis

spaske00 commented on app/resources/shaders/basic.glsl:
```diff
@@ -0,0 +1,35 @@
+uniform sampler2D texture_diffuse1;
+
+void main() {
+    FragColor = vec4(texture(texture_diffuse1, TexCoords).rgb, 1.0);
```
[CRITICAL] Fali implementacija osvetljenja scene u svim sejderima. Mislim da bi za ovu scenu PointLight bilo najprikladnije s` obzirom da već postoji lampa na sceni.

spaske00 reviewed (commented):
Projekat koji ne implementira osvetljenje na elementima scene ne ispunjava minimum potrebnih uslova za prolaz. Molim Vas, dodajte osvetljenje. 

Takođe, scena se odseca ako se kamera udalji. Trebalo bi da scena bude vidljiva i kada se kamera odalji tako da je celu obuhvati.

Razrešite prvo ove komentare pa mi javite da pogledam ponovo.

davidcuruvija committed:
Fixed the comments:
- .cpp files moved from include to src
- rand() replaced with <random>
- changed the distance of the far plane
- modified PROJECT-DESCRIPTION-TEMPLATE
- fixed lights

davidcuruvija committed:
Fixed normals

davidcuruvija committed:
- Added Spot light (camera is the light source)
- Fixed some issues

davidcuruvija committed:
- Deleting old files with no use

spaske00 commented:
Molim Vas, popunite i https://github.com/davidcuruvija/matf-rg-project-2024/pull/1#discussion_r2238987622. 
Ostatak izgleda dobro.

davidcuruvija committed:
Fixed a comment: PROJECT-DESCRIPTION-TEMPLATE modified

davidcuruvija commented:
Super, hvala! 
Želeo bih još samo da proverim koliko sam poena ostvario na projektu.

spaske00 commented:
25 bodova, rezultati uneti u tabelu.
